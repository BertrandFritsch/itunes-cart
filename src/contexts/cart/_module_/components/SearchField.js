// @flow

import React from 'react';

type Props = {

  /**
   * Notify the user wants to submit the search string
   */
  onSubmitSearchText: (searchText: string) => void
};

/**
 * CartHeader Component
 *
 * Use a pure component -- it doesn't re-render if neither its props or state have changed
 *
 * The search text is hold by the underlying input element
 */
export default class SearchField extends React.PureComponent<void, Props, void> {
  props: Props;

  /**
   * most recently entered search text, to avoid dispatching unchanged search text
   */
  searchText: string;

  handleChange = (newSearchText: string) => {
    if (this.searchText !== newSearchText) {
      this.props.onSubmitSearchText(this.searchText = newSearchText);
    }
  };

  render() {
    return (
      <input
          type="text"
          placeholder="Search a track..."
          onChange={ e => this.handleChange(e.target.value) }
      />
    );
  }
}
