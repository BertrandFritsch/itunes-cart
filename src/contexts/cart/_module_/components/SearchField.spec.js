import React from 'react';
import SearchField from './SearchField';

describe('SearchField', () => {
  it('should render correctly', () => {
    const props = {
      onSubmitSearchText: () => {}
    };

    const wrapper = mount(
      <SearchField { ...props } />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should dispatch the search text', () => {
    const props = {
      onSubmitSearchText: jest.fn()
    };

    const component = mount(
      <SearchField { ...props } />
    );

    component.find('input').simulate('change', { target: { value: 'pattern' } });
    expect(props.onSubmitSearchText).toBeCalledWith('pattern');
  });
});
