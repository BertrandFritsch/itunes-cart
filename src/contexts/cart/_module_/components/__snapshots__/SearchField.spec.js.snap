// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField should render correctly 1`] = `
<SearchField
  onSubmitSearchText={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search a track..."
    type="text"
  />
</SearchField>
`;
